net/udp/udp_send.c

```diff
@@ -115,15 +116,27 @@ void udp_send(FAR struct net_driver_s *dev, FAR struct udp_conn_s *conn)
               raddr = conn->u.ipv4.raddr;
             }
 
+#ifdef CONFIG_NET_IGMP
+          if (IN_MULTICAST(NTOHL(raddr)) &&
+              net_ipv4addr_cmp(dev->d_ipaddr, INADDR_ANY))
```

Comment:
why check d_ipaddr?

Reply:
It refers to linux(5.19)code  ,udp.c
![image](https://github.com/apache/nuttx/assets/129070278/e4970721-6ff5-48a6-a6b3-685ac0652de0)